src/Platform/Microsoft.Testing.Platform/TestHostControllers/TestHostControllersManager.cs

```diff
@@ -251,10 +251,10 @@ internal async Task<TestHostControllerConfiguration> BuildAsync(ServiceProvider
                 var instance = (IExtension)compositeFactoryInstance.GetInstance(serviceProvider);
 
                 // Check if we have already extensions of the same type with same id registered
-                if (dataConsumers.Any(x => x.Consumer.Uid == instance.Uid))
+                (IExtension consumer, int order)? existingConsumer = dataConsumers.Where(x => x.Consumer.Uid == instance.Uid).FirstOrDefault();
```

Comment:
We also want to throw a useful localized message if there are more than one results.